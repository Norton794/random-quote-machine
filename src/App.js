import React from 'react';


class App extends React.Component{
  constructor(props) {
     super(props);
    this.state = {
      random: 0,
      quotes: [
        {
          text: "Do what thou wilt shall be the whole of the Law.",
          author: "Aleister Crowley"
        },
        {
          text: "Be yourself; everyone else is already taken.",
          author: "Oscar Wilde"
        },
        {
          text: "I've learned that people will forget what you said, people will forget what you did, but people will never forget how you made them feel.",
          author: "Maya Angelou"
        },
        {
          text: "Live as if you were to die tomorrow. Learn as if you were to live forever.",
          author: "Mahatma Gandhi"
        },
        {
          text: "To be yourself in a world that is constantly trying to make you something else is the greatest accomplishment.",
          author: "Ralph Waldo Emerson"
        },
        {
          text: "It always seems impossible until it's done.",
          author: "Nelson Mandela"
        },
        {
          text: "You are the sum total of everything you've ever seen, heard, eaten, smelled, been told, forgot - it's all there. Everything influences each of us, and because of that I try to make sure that my experiences are positive.",
          author: "Maya Angelou"
        },
        {
          text: "By three methods we may learn wisdom: First, by reflection, which is noblest; Second, by imitation, which is easiest; and third by experience, which is the bitterest.",
          author: "Confucious"
        },
        {
          text: "Wonder is the beginning of wisdom.",
          author: "Socrates"
        },
        {
          text: "Everything that irritates us about others can lead us to an understanding of ourselves.",
          author: "Carl Gustav Jung"
        }
    ],
     classes: ["class-0", "class-1", "class-2", "class-3", "class-4", "class-5", "class-6", "class-7", "class-8", "class-9"]
    }
    
    this.changeQuote = this.changeQuote.bind(this)
  }
   
   
   componentDidMount() {
     setTimeout(()=>{
       this.setState({
         random: Math.floor(Math.random() * 10)
       }) 
     }, 200)
    }
   
   changeQuote(){
     setTimeout(()=>{
 
       this.setState({
         random: Math.floor(Math.random() * 10)
       }) 
 
     }, 300)
   }
   
    
   
   render(){
     return(
       <div id="quote-box" class={this.state.classes[this.state.random]}>
         <h3 id="text">{this.state.quotes[this.state.random].text}</h3>
         <p id="author"> - {this.state.quotes[this.state.random].author}</p>
         <div id="group-buttons">
           <button id="new-quote" onClick={this.changeQuote}>New Quote</button>
           <a id="tweet-quote" href="twitter.com/intent/tweet">Tweet Quote</a>
         </div> 
       </div>
     );
   }
     
   
 }
 

export default App;
