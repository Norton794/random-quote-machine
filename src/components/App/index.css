* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
}

body {
  background-color: #efefef;
  color: #1e96fc;
  padding-top: 30vh;
  font-size: 10px;
  font-family: sans-serif;
}

a {
  text-decoration: none;
}

#quote-box {
  width: 50%;
  margin: 0 auto;
  padding: 2.5em 2.5em;
  border-radius: 0.7em;
  background-color: #1e96fc;
  color: #fff;
  font-weight: 600;
}

#text {
  font-size: 2.8em;
  margin-bottom: 1.2em;
  text-align: center;
  line-height: 1.5em;
}

#author {
  font-size: 1.8em;
  margin-bottom: 1.2em;
  text-align: right;
}

#new-quote {
  background-color: #fff;
  border: none;
  padding: 12px 12px;
  color: #1e96fc;
  cursor: pointer;
  font-size: 1.3em;
  font-weight: bold;
  opacity: 0.95;
  border-radius: 20px;
}

#group-buttons {
  display: flex;
  justify-content: center;
  padding-top: 1.6em;
}

#new-quote:hover{
  opacity: 1;
}
