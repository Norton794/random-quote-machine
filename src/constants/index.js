export const quotes = [
    {
      text: "Do what thou wilt shall be the whole of the Law.",
      author: "Aleister Crowley"
    },
    {
      text: "Be yourself; everyone else is already taken.",
      author: "Oscar Wilde"
    },
    {
      text: "I've learned that people will forget what you said, people will forget what you did, but people will never forget how you made them feel.",
      author: "Maya Angelou"
    },
    {
      text: "Live as if you were to die tomorrow. Learn as if you were to live forever.",
      author: "Mahatma Gandhi"
    },
    {
      text: "To be yourself in a world that is constantly trying to make you something else is the greatest accomplishment.",
      author: "Ralph Waldo Emerson"
    },
    {
      text: "It always seems impossible until it's done.",
      author: "Nelson Mandela"
    },
    {
      text: "You are the sum total of everything you've ever seen, heard, eaten, smelled, been told, forgot - it's all there. Everything influences each of us, and because of that I try to make sure that my experiences are positive.",
      author: "Maya Angelou"
    },
    {
      text: "By three methods we may learn wisdom: First, by reflection, which is noblest; Second, by imitation, which is easiest; and third by experience, which is the bitterest.",
      author: "Confucious"
    },
    {
      text: "Wonder is the beginning of wisdom.",
      author: "Socrates"
    },
    {
      text: "Everything that irritates us about others can lead us to an understanding of ourselves.",
      author: "Carl Gustav Jung"
    }
  ];
  
  
  export const pickAndChoose = () => {
    return Math.floor(Math.random() * 10);
  }