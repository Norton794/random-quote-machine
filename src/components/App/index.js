import {Component} from 'react';
import './index.css';
import {quotes, pickAndChoose} from '../../constants/index';



class App extends Component {
  constructor() {
    super();
    this.state = {
      random: 0,
      quotes,
    }

    this.changeQuote = this.changeQuote.bind(this)
  }


  componentDidMount() {
    this.setState({
      random: pickAndChoose()
    })

  }

  changeQuote() {
    this.setState({
      random: pickAndChoose()
    })

  }



  render() {
    const { random, quotes } = this.state;
    return (
      <div id="quote-box">
        <h3 id="text">{quotes[random].text}</h3>
        <p id="author"> - {quotes[random].author}</p>
        <div id="group-buttons">
          <button id="new-quote" onClick={this.changeQuote}>New Quote</button>
        </div>
      </div>
    );
  }


}


export default App;