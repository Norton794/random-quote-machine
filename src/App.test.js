import React from 'react';
import ReactDOM from 'react-dom';
import App from '../src/components/App/index';
import renderer from 'react-test-renderer';
import {pickAndChoose} from '../src/constants/index';

describe('App', () => {
  it('renderização', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('snapshot', ()=>{
    const component = renderer.create(
      <App />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('random', ()=>{
    const value = pickAndChoose();
    expect(value).toBeTruthy();
  });

  test('type', ()=>{
    const value = pickAndChoose();
    expect(typeof value).toBe("number");
  });
});

