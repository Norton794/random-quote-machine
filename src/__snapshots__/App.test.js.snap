// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<div
  id="quote-box"
>
  <h3
    id="text"
  >
    Everything that irritates us about others can lead us to an understanding of ourselves.
  </h3>
  <p
    id="author"
  >
     - 
    Carl Gustav Jung
  </p>
  <div
    id="group-buttons"
  >
    <button
      id="new-quote"
      onClick={[Function]}
    >
      New Quote
    </button>
  </div>
</div>
`;
